config/config.go

```diff
@@ -305,6 +306,7 @@ func LoadReceiver(componentConfig *viper.Viper, typeStr configmodels.Type, fullN
 	// Create the default config for this receiver.
 	receiverCfg := factory.CreateDefaultConfig()
 	receiverCfg.SetName(fullName)
+	expandEnvLoadedConfig(receiverCfg)
```

Comment:
Shouldn't this be done in `loadReceivers` instead of here? It looks odd to me that this is different that for the rest of component kinds

Reply:
This is because receivers are handled differently since #658, it's by design. Consider this solved.